### PR TITLE
Add opentelemetry test for SystemApicast

### PR DESCRIPTION
**Pull request by mdujava**
It would be nice to get merged into 2.14

### CODE REVIEW

File: testsuite/gateways/apicast/operator.py [hunk 1]
```diff
@@ -216,6 +216,12 @@ def _update(apicast):
         self.apicast.modify_and_apply(_update)
         self.reload()
 
+    def disconnect_open_telemetry(self, *args):
+        """
+        Current tests delete apicast after opentelemetry tests, cleanup of the
+        apicast itself will clean all configuration for opentelemetry as well
+        """
```
Comment (dhlavac) [inline testsuite/gateways/apicast/operator.py:223]: Shouldn't there be at least `pass` ?

File: testsuite/tests/apicast/parameters/jaeger/test_open_telemetry_apicast_integration.py [hunk 2]
```diff
@@ -7,14 +7,13 @@
 import pytest
 
 from packaging.version import Version  # noqa # pylint: disable=unused-import
-from testsuite import APICAST_OPERATOR_VERSION  # noqa # pylint: disable=unused-import
+from testsuite import TESTED_VERSION, APICAST_OPERATOR_VERSION  # noqa # pylint: disable=unused-import
```
Comment (dhlavac) [inline testsuite/tests/apicast/parameters/jaeger/test_open_telemetry_apicast_integration.py:10]: The version check was removed, why do we need `TESTED_VERSION` ?

File: testsuite/gateways/apicast/operator.py (hunk 1 above)
Comment (mdujava) [reply testsuite/gateways/apicast/operator.py:223]: doc string is a valid content of a empty function by itself
https://pylint.pycqa.org/en/v3.0.3/user_guide/messages/warning/unnecessary-pass.html

**Review by dhlavac — APPROVED** [review]

**Review by mkudlej — COMMENTED** [review]
LGTM

**Review by mkudlej — APPROVED** [review]

**Pull request merged**